CRVResponse/src/CrvPlot_module.cc

```diff
@@ -136,35 +133,41 @@ namespace mu2e
     std::vector<TGraph*> graphVector;
     std::vector<TGaxis*> axisVector;
 
-    double extension = (_timeEnd-_timeStart)*0.1;
-
-    for(int SiPM=0; SiPM<4; SiPM++)
-    {
-      _canvas[_icanvas]->cd(SiPM+1);
-
 //Photon Times
-      std::vector<double> photonTimesAdjusted;
-      if(photons!=crvPhotonsCollection->end())
+    std::vector<double> photonTimesAdjusted[4];
+    double histMax[4]={0};
+    CrvPhotonsCollection::const_iterator photons;
+    for(photons=crvPhotonsCollection->begin(); photons!=crvPhotonsCollection->end(); photons++)
+    {
+      if(photons->GetScintillatorBarIndex()==barIndex)
       {
-        const std::vector<CrvPhotons::SinglePhoton> &photonTimes = photons->second.GetPhotons(SiPM);
+        int SiPM=photons->GetSiPMNumber();
+        if(SiPM<0 || SiPM>3) continue;
```

Comment:
Is this possible?  Is it an error condition?

Reply:
It should not happen, and none of the existing Crv modules would create a photons object with an invalid SiPM number. So perhaps this test is not necessary.

Reply:
If you think it could only happen due to memory corruption, then you should throw instead of just continue.

Reply:
I wasn't thinking of memory corruption, I thought there could be a module which puts incorrect data into the event. It should throw in either case. I will make the change.

Reply:
For future reference: I agree with Dave that no code should just continue on an error.  The choices available now are to throw an exception or to log that the error has happened.   For now, I think that throwing an exception is almost always the right choice.  Code that will run in the trigger should not throw exceptions - for now the only option is to log the error.  One can also imagine that the TDAQ group will provides the functionality for code to raise an alarm and continue.  I don't know the trigger group's plan for logging errors, inspecting the logs and raising alarms.